contracts/oracle/src/contract/alarms.rs

```diff
@@ -89,33 +99,31 @@ impl MarketAlarms {
         )
     }
 
-    pub fn try_notify_alarms<BaseC>(
-        storage: &mut dyn Storage,
+    pub fn try_notify_alarms<'a, BaseC>(
+        &mut self,
+        storage: &dyn Storage,
         mut batch: Batch,
-        prices: &[SpotPrice],
+        prices: impl Iterator<Item = BasePrice<BaseC, SwapGroup>> + 'a,
         max_count: u32, // TODO: type alias
     ) -> Result<Response, ContractError>
     where
         BaseC: Currency,
     {
-        let mut next_id = Self::MSG_ID.may_load(storage)?.unwrap_or_default();
-        let initial_id = next_id;
+        let initial_id = self.0;
 
         Self::alarms_iter::<BaseC>(storage, prices)
             .take(max_count.try_into()?)
-            .try_for_each(|addr| Self::schedule_alarm(&mut batch, addr?, &mut next_id))?;
-
-        Self::MSG_ID.save(storage, &next_id)?;
+            .try_for_each(|addr| Self::schedule_alarm(&mut batch, addr?, &mut self.0))?;
```

Comment:
[everywhere] to improve readability pls do not use so generic argument names like `addr`. `Addr` is a type. In this particular case, we may name the reference `receiver` or `listener`.